server/src/main/java/com/datastrato/gravitino/server/web/rest/PermissionOperations.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.server.web.rest;
+
+import com.codahale.metrics.annotation.ResponseMetered;
+import com.codahale.metrics.annotation.Timed;
+import com.datastrato.gravitino.GravitinoEnv;
+import com.datastrato.gravitino.authorization.AccessControlManager;
+import com.datastrato.gravitino.dto.requests.RoleGrantRequest;
+import com.datastrato.gravitino.dto.responses.GrantResponse;
+import com.datastrato.gravitino.dto.responses.RevokeResponse;
+import com.datastrato.gravitino.metrics.MetricNames;
+import com.datastrato.gravitino.server.web.Utils;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+
+@Path("/metalakes/{metalake}/permissions")
+public class PermissionOperations {
+
+  private final AccessControlManager accessControlManager;
+
+  @Context private HttpServletRequest httpRequest;
+
+  public PermissionOperations() {
+    // Because accessManager may be null when Gravitino doesn't enable authorization,
+    // and Jersey injection doesn't support null value. So PermissionOperations chooses to retrieve
+    // accessControlManager from GravitinoEnv instead of injection here.
+    this.accessControlManager = GravitinoEnv.getInstance().accessControlManager();
+  }
+
+  @POST
+  @Path("users/{user}/roles/")
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "grant-role-to-user." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "grant-role-to-user", absolute = true)
+  public Response grantRoleToUser(
+      @PathParam("metalake") String metalake,
+      @PathParam("user") String user,
+      RoleGrantRequest request) {
```

Comment:
I think it would be better to support grant or revoke multiple roles to user/group, not just one role, like what we do for table changes, WDYT? @qqqttt123

Reply:
Done.